src/exit_codes.rs

```diff
@@ -7,9 +7,21 @@ pub enum ExitCode {
 impl Into<i32> for ExitCode {
     fn into(self) -> i32 {
         match self {
-            ExitCode::Success => 0,
-            ExitCode::GeneralError => 1,
-            ExitCode::KilledBySigint => 130,
+            Self::Success => 0,
+            Self::GeneralError => 1,
+            Self::KilledBySigint => 130,
         }
     }
 }
+
+impl ExitCode {
+    pub fn error_if_any_error(results: Vec<Self>) -> Self {
+        if results.iter().any(|s| match s {
```

Comment:
Maybe this `match` should be extracted into another function: `ExitCode::is_error`?

Reply:
This way, we could simply run: `if results.iter().any(ExitCode::is_error) { … }`

Reply:
😁👍